### PR TITLE
Expose `batch_size` parameter for `rg.load`

### LINKED ISSUE #2454: Introduce a `batch_size` argument for `rg.load`
**Is your feature request related to a problem? Please describe.**
I'd like to be able to specify a `batch_size` argument in `rg.load` to have more influence over potential timeouts.

**Describe the solution you'd like**
I'd like to be able to do:
```python
data = rg.load(name, batch_size=64)
```

**Describe alternatives you've considered**
Sticking with the current behaviour, i.e. using a batch size of 250 at all times.

**Additional context**
See https://github.com/argilla-io/argilla/pull/2434#issuecomment-1448303805

### PR DESCRIPTION
**Pull request by tomaarsen**
Closes #2454 

Hello!

## Pull Request overview
* Extend my previous work to expose a `batch_size` parameter to `rg.load` (and `rg.scan`, `_load_records_new_fashion`).
  * Update the docstrings for the public methods.
  * The actual implementation is very simple due to my earlier work in #2434:
    * `batch_size = self.DEFAULT_SCAN_SIZE` -> `batch_size = batch_size or self.DEFAULT_SCAN_SIZE`
* Add an (untested!) warning if `batch_size` is supplied with an old server version.
  * (We don't have compatibility tests for old server versions)
* I've simplified `test_scan_efficient_limiting` with `batch_size` rather than using a mocker to update `DEFAULT_SCAN_SIZE`.
  * I've also extended this test to test `rg.load` in addition to `active_api().datasets.scan`.

## Note
Unlike previously discussed with @davidberenstein1957, I have *not* added a warning like so:
https://github.com/argilla-io/argilla/blob/f5834a5408051bf1fa60d42aede6b325dc07fdbd/src/argilla/client/client.py#L338-L343
The server limits the maximum batch size to 500 for loading, so there is no need to have a warning when using a batch size of over 5000.

## Discussion
1. Should I include in the docstring that the default batch size is 250? That would be "hardcoded" into the docs, so if we ever change the default (`self.DEFAULT_SCAN_SIZE`), then we would have to remember to update the docs too.
2. Alternatively, should we deprecate `self.DEFAULT_SCAN_SIZE` and enforce that `batch_size` must be set for `datasets.scan`, with a default size of 250 everywhere? Then people can see the default batch size in the signature?

I think we should do option 2. Let me know how you feel.

---

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

Modified and updated a test, ran all tests.

**Checklist**

- [x] I have merged the original branch into my forked branch
- [x] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [x] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

---

- Tom Aarsen

### CODE REVIEW

File: src/argilla/client/api.py [hunk 1]
```diff
@@ -206,6 +206,7 @@ def load(
     ids: Optional[List[Union[str, int]]] = None,
     limit: Optional[int] = None,
     id_from: Optional[str] = None,
+    batch_size: Optional[int] = None,
```
Comment (frascuchon) [inline src/argilla/client/api.py]: Maybe it makes sense to have an explicit default value here. 
`batch_size: int = 250`

File: src/argilla/client/client.py [hunk 2]
```diff
@@ -495,6 +497,13 @@ def load(
                 f"`pip install argilla=={err.api_version}`",
                 category=UserWarning,
             )
+            if batch_size is not None:
```
Comment (frascuchon) [inline src/argilla/client/client.py:500]: Nice catch!

**Review by frascuchon — APPROVED** [review]

File: src/argilla/client/api.py (hunk 1 above)
Comment (tomaarsen) [reply src/argilla/client/api.py]: Will do, I'll use 250 throughout 👍

**Comment by davidberenstein1957** [comment]
@frascuchon is the 500 max size a ES limit or an Argilla limit? 
If Argilla limit:
What happens if we increase it? If the python client defaults to 250, I would once again argue that it might be worth increasing/removing the server-side limit and letting this be configurable by the user. Most clients will just use the defaults and clients that want to scale, want to be able to configure this and potentially assign extra resources.

File: src/argilla/client/api.py (hunk 1 above)
Comment (tomaarsen) [reply src/argilla/client/api.py]: Done

**Comment by frascuchon** [comment]
@davidberenstein1957 It's an argilla endpoint limit. For elasticsearch will be set normally to 10000. Yes, we can increase this limit once the batch_size is exposed in `rg.load`. @tomaarsen could you include the change in this PR? The value to change is here https://github.com/argilla-io/argilla/blob/develop/src/argilla/server/apis/v0/handlers/records_search.py#L60

We could start  by aligning it with the current pagination value for search (1000)

**Comment by tomaarsen** [comment]
@frascuchon Done, it's at 1000 now

**Review by frascuchon — APPROVED** [review]

**Pull request merged**